algorithms/src/CMakeLists.txt

```diff
@@ -7,18 +7,24 @@ KOKKOS_INCLUDE_DIRECTORIES(${CMAKE_CURRENT_SOURCE_DIR})
 
 #-----------------------------------------------------------------------------
 
-FILE(GLOB HEADERS *.hpp)
-FILE(GLOB SOURCES *.cpp)
-LIST(APPEND HEADERS ${CMAKE_CURRENT_BINARY_DIR}/${PACKAGE_NAME}_config.h)
+FILE(GLOB ALGO_HEADERS *.hpp)
+FILE(GLOB ALGO_SOURCES *.cpp)
+LIST(APPEND ALGO_HEADERS ${CMAKE_CURRENT_BINARY_DIR}/${PACKAGE_NAME}_config.h)
+
+INSTALL (
+  DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/"
+  DESTINATION ${KOKKOS_HEADER_DIR}
+  FILES_MATCHING PATTERN "*.hpp"
+)
```

Comment:
Can you elaborate why you preferred doing this directly in the sub package CMakeLists.txt files rather than delegating to the `KOKKOS_ADD_INTERFACE_LIBRARY` or `KOKKOS_ADD_LIBRARY ` functions?